lib/IRGen/GenDecl.cpp

```diff
@@ -1341,6 +1338,36 @@ void IRGenerator::emitDynamicReplacements() {
 
   auto &IGM = *getPrimaryIGM();
 
+  // Collect all the type metadata accessor replacements.
+  SmallVector<OpaqueTypeArchetypeType *, 8> newFuncTypes;
+  SmallVector<OpaqueTypeArchetypeType *, 8> origFuncTypes;
+  for (auto *newFunc : DynamicReplacements) {
+    if (!newFunc->getLoweredFunctionType()->hasOpaqueArchetype())
```

Comment:
This should be a structural walk over the SILFunctionType using `newFunc->getLoweredFunctionType().each()`